### PR TITLE
moderator notification setting ui

### PR DESCRIPTION
**Pull request by abeglova**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [x] Mobile width screenshots
  - [x] Tag @ferdi or @pdpinch for review
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested
Screen Shot 2020-12-14 at 7 59 09 AM

#### What are the relevant tickets?
closes https://github.com/mitodl/open-discussions/issues/3244

#### What's this PR do?
This PR adds an option in /settings/notifications that is only visible to moderators to allow moderators to individually opt in or out of notifications

#### How should this be manually tested?
Run `docker-compose run web ./manage.py populate_notification_settings`

Log in as a non-moderator user and go to `http://localhost:8063/settings/`. There should be no changes

Log in as a moderator user and go to `http://localhost:8063/settings/`. There should be a new checkbox to allow moderator notifications.

Keep moderator notifications enabled in that ui and enable notifications in `http://localhost:8063/manage/c/edit/<channel>/basic/`

Make a post in the channel. Then run

```
from notifications.tasks import *
send_unsent_email_notifications()
```
You should get a notification on the MAILGUN_RECIPIENT_OVERRIDE email

If moderator notifications are turned off for either the user or the channel, you should not receive a notification email

### Updated you based on @Ferdi's comment:
Screen Shot 2020-12-22 at 2 11 34 PM

### Updated UI based on additional comment from @Ferdi:
Screen Shot 2021-01-11 at 9 02 17 PM

### CODE REVIEW

File: notifications/views.py [hunk 1]
```diff
@@ -27,5 +29,14 @@ class NotificationSettingsViewSet(
...
+
         user = self.request.user
-        return user.notification_settings
+
+        if admin_api.is_moderator_for_any_channel(user.username):
+            return user.notification_settings
+        else:
+            return user.notification_settings.exclude(
+                notification_type=NOTIFICATION_TYPE_MODERATOR
+            )
```
Comment (mbertrand) [inline notifications/views.py:42]: Is this part necessary, would a non-moderator not have that notificiation type?

**Comment by Ferdi** [comment]
looking at the screenshot , it's not clear what this does .  
If i'm guessing right what the intent is , this should work something like : 

When do you want to receive notification ?
[ ] when someone responds to my post / comment 
[ ] when someone post to a channel I moderate

**Review by mbertrand — CHANGES_REQUESTED** [review]
Functionality works great! 

One thing I noticed not related to this PR is that `docker-compose run web ./manage.py populate_notification_settings` didn't do anything on my dev instance without the `--inactive` flag because `Profile.last_active_on` was null for all my users.  The same seems to be true on RC and production, including my own account which I just logged in today.  I think it's probably a bug?

File: notifications/views.py (hunk 1 above)
Comment (abeglova) [reply notifications/views.py:42]: Yes, we do not delete the notification setting if a user is removed as a moderator

**Comment by Ferdi** [comment]
@abeglova even better would be something like this : 
When do you want to receive notification ?
[ ] when someone responds to my post / comment
[ ] when someone post to a channel I moderate
       [ ] channel 1
       [ ] channel 2 
       [ ] etc
       
The "when someone post to a channel I moderate" shows up only when user is moderating 1 ore more channels  AND those channels have the settings turned on in the setting page

File: notifications/views.py (hunk 1 above)
Comment (abeglova) [reply notifications/views.py:42]: I updated the code to remove the moderator setting when a user is removed as a moderator

**Comment by abeglova** [comment]
Back to you @mbertrand. To test, delete all your local NotificationSetting objects with notification_type="moderator_posts" and run `populate_notification_settings` again

**Review by mbertrand — APPROVED** [review]
Unfortunately I'm not able to test the functionality right now due to ongoing laptop issues, but the code looks good so approving based on that.

**Review by mbertrand — APPROVED** [review]
Functionality works great too, though the filtering by inactive users is still a problem because of https://github.com/mitodl/open-discussions/issues/3266

**Pull request merged**